RS: 2020Q1: Virtual keyboard

alekseyvlivanov opened the pull request:
1. Task - https://github.com/rolling-scopes-school/tasks/blob/master/tasks/codejam-virtual-keyboard.md
2. Screenshot - 
![image](https://user-images.githubusercontent.com/14129152/78750549-d0f44e00-79b3-11ea-8b10-615172ee9a57.png)
3. Solution - https://alekseyvlivanov.github.io/virtual-keyboard/index.html
4. Submitted:  2020-04-06 / Deadline: 2020-04-07
5. Total - all requirements are done
**Task (9 / 9)**
Максимальный балл за задание: **110**

_Минимальный набор:_
- реализована генерация DOM-элементов и body в index.html пустой изначально (может находится только тег script): **+20**
- нажатие на кнопку на физической клавиатуре подсвечивает кнопку на виртуальной(проверять следуюет нажатие цифр, букв, знаков припинания, backspace, del(если она присутствует), enter, shift, alt, ctrl, tab, caps lock, space, стрелки вниз-вверх-влево-вправо): **+10**

_Стандартный набор:_
- есть переключение между русским и английским языком(сочетание клавиш для переключения языка должно быть указано на странице с клавиатурой), а так же сохранение выбранного языка: **+15**
- клики мышкой по кнопкам на виртуальной клавиатуре или нажатие на кнопки физической клавиатуры, выводят символы в инпут(textarea): **+15**

_Дополнительный набор:_
- реализована анимация нажатия на кнопку: **+15**

_Технические требования:_
- использование в коде фишек стандарта ES6 и выше (classes, деструктуризация и т.д.): **+15**
- использование eslint: **+10**
- требования к репозиторию, коммитам и PR выполнены: **+10**

alekseyvlivanov committed:
feat: add basic project structure

alekseyvlivanov committed:
feat: add basic DOM structure

alekseyvlivanov committed:
feat: add layout for ENG

alekseyvlivanov committed:
feat: add animation for key press

alekseyvlivanov committed:
feat: add switching layouts

alekseyvlivanov committed:
feat: add mouse click simulation for key press

alekseyvlivanov committed:
fix: change keyboard combination for switching layouts to avoid sticking

alekseyvlivanov committed:
feat: add full solution

alekseyvlivanov committed:
Merge branch 'master' into dev

alekseyvlivanov committed:
feat: add shift char for Period in russian layout

Jammmes commented on script.js:
```diff
@@ -0,0 +1,274 @@
+        return;
+      }
+
+      if (e.code === 'CapsLock' && !e.repeat) {
```
Здесь и дальше предпочтительно было бы сравнивать e.code не со строкой, а с константой, содержащей строку

Jammmes commented on script.js:
```diff
@@ -0,0 +1,274 @@
+      } else {
+        key.classList.add('active');
+
+        if ((e.ctrlKey || e.metaKey) && e.altKey && !e.repeat) {
```
В случае, если перечисляется более 2-3 вариантов, все же предпочтительнее использовать конструкцию switch case вместо if /else if

Jammmes commented on script.js:
```diff
@@ -0,0 +1,274 @@
+    Array.from(this.keyboard.querySelectorAll('.keyboard__key')).forEach(
+      (e) => {
+        if (!keyboardKeys[e.id].func) {
+          if (e.id === 'Backquote' && this.lang === 'en') {
```
Тот же случай - много if/else if - лучше сделать switch case

dependabot[bot] committed:
build(deps): bump lodash from 4.17.15 to 4.17.20

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.15 to 4.17.20.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.15...4.17.20)

Signed-off-by: dependabot[bot] <support@github.com>

alekseyvlivanov committed:
Merge pull request #2 from alekseyvlivanov/dependabot/npm_and_yarn/lodash-4.17.20

build(deps): bump lodash from 4.17.15 to 4.17.20

alekseyvlivanov committed:
feat: add Netlify redirects

alekseyvlivanov committed:
Merge branch 'gh-pages' of github.com:alekseyvlivanov/virtual-keyboard into gh-pages
